auto-tuning/proxy/src/Proxy/Common.cpp

```diff
@@ -0,0 +1,150 @@
+    {Kernel::NeighborDR, "neigh_dr"},
+    {Kernel::GodunovDR, "godunov_dr"},
+    {Kernel::AllDR, "all_dr"},
+};
+
+const std::unordered_map<std::string, Kernel> InvMap{{"all", Kernel::All},
+                                                     {"local", Kernel::Local},
+                                                     {"neigh", Kernel::Neighbor},
+                                                     {"ader", Kernel::Ader},
+                                                     {"localwoader", Kernel::LocalWOAder},
+                                                     {"neigh_dr", Kernel::NeighborDR},
+                                                     {"godunov_dr", Kernel::GodunovDR},
+                                                     {"all_dr", Kernel::AllDR}};
+} // namespace
+
+namespace seissol::proxy {
+
+auto Aux::kernel2str(const std::vector<Kernel>& kernel) -> std::string {
+  std::string output;
+  for (std::size_t i = 0; i < kernel.size(); ++i) {
+    if (i > 0) {
+      output += ",";
+    }
+    if (Map.find(kernel[i]) != Map.end()) {
+      output += Map.at(kernel[i]);
+    } else {
+      throw std::runtime_error("Unknown kernel type");
+    }
+  }
+  return output;
+}
+
+auto Aux::str2kernel(const std::string& kernelStr) -> std::vector<Kernel> {
+  auto parts = utils::StringUtils::split(kernelStr, ',');
+  for (auto& part : parts) {
+    utils::StringUtils::trim(part);
+  }
+
+  std::vector<Kernel> kernels(parts.size());
+  for (std::size_t i = 0; i < parts.size(); ++i) {
+    if (InvMap.find(parts[i]) != InvMap.end()) {
+      kernels[i] = InvMap.at(parts[i]);
+    } else {
+      throw std::runtime_error("Unknown kernel string: " + parts[i]);
+    }
+  }
+
+  return kernels;
+}
+
+auto Aux::getAllowedKernels() -> std::vector<std::string> {
+  std::vector<std::string> kernels{};
+  kernels.reserve(InvMap.size());
+  for (const auto& kernel : InvMap) {
+    kernels.push_back(kernel.first);
+  }
+  return kernels;
+}
+
+void Aux::writeOutput(std::ostream& stream,
+                      const ProxyOutput& output,
+                      const std::string& kernelStr,
+                      OutputFormat format) {
+  if (format == OutputFormat::Plain) {
+    stream << "=================================================\n";
+    stream << "===            PERFORMANCE SUMMARY            ===\n";
+    stream << "=================================================\n";
+    stream << "seissol proxy mode                  : " << kernelStr << '\n';
+    stream << "time for seissol proxy              : " << output.time << '\n';
+    stream << "cycles                              : " << output.cycles << '\n';
+    stream << '\n';
+    stream << "GFLOP (libxsmm)                     : " << output.libxsmmNumTotalGFlop << '\n';
+    stream << "GFLOP (pspamm)                      : " << output.pspammNumTotalGFlop << '\n';
+    stream << "GFLOP (libxsmm + pspamm)            : " << output.libxsmmAndpspammNumTotalGFlop
+           << '\n';
```

Comment:
They _do_ still have an effect if you compile with Debug. See e.g. https://github.com/SeisSol/SeisSol/actions/runs/11748077954/job/32731391066#step:7:27 .

However, you may be right with regard to the `FLOPs/cycle` etc. statements. The `cycle` is currently not implemented anymore, so it would make sense disabling it for now.

What we could do in addition would be to just remove those fields in non-debug mode—I'm not 100%ly certain about that one though.